esphome/components/status/binary_sensor.py

```diff
@@ -5,17 +5,28 @@
     DEVICE_CLASS_CONNECTIVITY,
     ENTITY_CATEGORY_DIAGNOSTIC,
 )
+from . import CONF_STATUS_ID
```

Comment:
Im not sure why you didnt just add this, nothing else in this PR is required. And will force either wifi or ethernet to be present when trying to use this binary sensor.

```suggestion

DEPENDENCIES = ["network"]
```